src/client/Microsoft.Identity.Client/Instance/Authority.cs

```diff
@@ -208,6 +218,22 @@ private static void ValidateSameHost(AuthorityInfo requestAuthorityInfo, Authori
             }
         }
 
+        private static bool IsAuthorityAliased(RequestContext requestContext, AuthorityInfo requestAuthorityInfo)
+        {
+            var instanceDiscoveryManager = requestContext.ServiceBundle.InstanceDiscoveryManager;
+            var result = instanceDiscoveryManager.GetMetadataEntryAsync(requestContext.ServiceBundle.Config.AuthorityInfo, requestContext).Result;
```

Comment:
We should not do this. This will block the thread. Never use `.Result` in production code. This is an async call, so you must make all the methods in the call chain async. There aren't that many.

Reply:
makes sense